### PR TITLE
Add a pseudo console as a rudimentary API for player interaction

### PR DESCRIPTION
**Pull request by nopeless**
Listening for keys is a good idea but having a text input seems even better.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,136 @@
...
+
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
 node_modules/
-dist/
```
Comment (Cazka) [inline .gitignore:2]: Can we please keep this in .gitignore. Build files don't need to be in the repo.

File: examples/in_game_console.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+// ==/UserScript==
+
+if (!window.diepAPI) return window.alert('Please install diepAPI to use this script');
+
+const { player, game, entityManager, Vector } = window.diepAPI;
+
+
+const id = "diep-ingame-console";
+
+`
```
Comment (Cazka) [inline examples/in_game_console.js:19]: Did you forgot something here?

File: examples/in_game_console.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    cursor: move;
+}
+
+</style>
+<div id="${id}">
+</div>
+`
+
+// Make the DIV element draggable:
+dragElement(document.getElementById("mydiv"));
```
Comment (Cazka) [inline examples/in_game_console.js:35]: `document.getElementById("mydiv")` will always be **null**

**Review by Cazka — COMMENTED** [review]
Thanks for your first PR.

I've left some comments for you.

And I'm not sure if this should be added to diepAPI, Could you provide a good use case for this feature?

File: .gitignore (hunk 1 above)
Comment (nopeless) [reply .gitignore:2]: Will fix

File: examples/in_game_console.js (hunk 2 above)
Comment (nopeless) [reply examples/in_game_console.js]: yes I was working on a hotfix html file and forgot to update the userscript file. It is reflected in https://github.com/Cazka/diepAPI/pull/26/commits/1d2459a7f01b12892a27b61c9fd0b5c78efbeed2

File: examples/in_game_console.js (hunk 3 above)
Comment (nopeless) [reply examples/in_game_console.js]: Same comment as above. Fix reflected in https://github.com/Cazka/diepAPI/pull/26/commits/1d2459a7f01b12892a27b61c9fd0b5c78efbeed2

**Comment by nopeless** [comment]
Thanks for your review

**Comment by nopeless** [comment]
Meanwhile, can we discuss about the UserConsole API? I would like to add some sort of global command handler. I would like some future visions from you so we can coordinate our projects

**Comment by nopeless** [comment]
image
this is how I use it

**Comment by nopeless** [comment]
@Cazka I'm looking at the canvas kit api right now

Is there a way to "dump" every canvas call? So I can map it

**Comment by nopeless** [comment]
@Cazka bump

**Comment by Cazka** [comment]
This looks cool but why do we need this in diepAPI. can't you just use the dev console?

I think that this is not relevant enough to be in diepAPI and should be implemented entirely as an userscript similar to [entity_debug_tool.user.js](https://github.com/Cazka/diepAPI/blob/main/examples/entity_debug_tool.user.js)

**Pull request closed without merging**